final_prep/mod_zero_hero.rb

```diff
@@ -1,52 +1,78 @@
 # Challenge - See if you can follow the instructions and complete the exercise in under 30 minutes!
 
 # Declare two variables - hero_name AND special_ability - set to strings
+hero_name = "Batman"
+special_ability = "rich"
 
 # Declare two variables - greeting AND catchphrase
 #   greeting should be assigned to a string that uses interpolation to include the hero_name
 #   catchphrase should be assigned to a string that uses interpolation to include the special_ability
+greeting = "I'm #{hero_name}."
+catchphrase = "I'm #{special_ability}."
 
 # Declare two variables - power AND energy - set to integers
+power = 100
+energy = 50
 
 # Declare two variables - full_power AND full_energy
 #   full_power should multiply your current power by 500
 #   full_energy should add 150 to your current energy
+full_power = power * 500
+full_energy = energy + 150
 
 # Declare two variables - is_human and identity_concealed - assigned to booleans
-
+is_human = true
+identity_concealed = true
 
 # Declare two variables - arch_enemies AND sidekicks
 #   arch_enemies should be an array of at least 3 different enemy strings
 #   sidekicks should be an array of at least 3 different sidekick strings
+arch_enemies = ["Joker", "Bane", "Two-Face"]
+sidekicks = ["Jason Todd", "Dick Grayson", "Tim Drake"]
 
 # Print the first sidekick to your terminal
+p sidekicks[0]
 
 # Print the last arch_enemy to the terminal
+p arch_enemies[-1]
 
 # Write some code to add a new arch_enemy to the arch_enemies array
+arch_enemies << "Mr. Freeze"
 
 # Print the arch_enemies array to terminal to ensure you added a new arch_enemey
+p arch_enemies
 
 # Remove the first sidekick from the sidekicks array
+sidekicks.shift
 
 # Print the sidekicks array to terminal to ensure you added a new sidekick
+p sidekicks
 
 # Create a function called assess_situation that takes three arguments - danger_level, save_the_day, bad_excuse
 #   - danger_level should be an integer
-#   - save_the_day should be a string a hero would say once they save the day 
+#   - save_the_day should be a string a hero would say once they save the day
 #   - bad_excuse should be a string a hero would say if they are too afraid of the danger_level
 
 # Your function should include an if/else statement that meets the following criteria
 #   - Danger levels that are above 50 are too scary for your hero. Any danger level that is above 50 should result in printing the bad_excuse to the terminal
 #   - Anything danger_level that is between 10 and 50 should result in printing the save_the_day string to the terminal
 #   - If the danger_level is below 10, it means it is not worth your time and should result in printing the string "Meh. Hard pass." to the terminal.
+def assess_situation(danger_level, save_the_day, bad_excuse)
+  if danger_level > 50
+    p bad_excuse
+  elsif danger_level >= 10 && danger_level <= 50
```

Comment:
This is great, but sometimes you can avoid some complex logic by being strategic with which condition you place in the else block.